impl-mutex.go

```diff
@@ -114,6 +114,14 @@ func (p *process) Go(f ProcessFunc) Process {
 	return child
 }
 
+// SetTeardown to assign a teardown function
```

Comment:
the doc should say that this must be called before process is added as a child or parent to other processes, because otherwise it is not threadsafe.